doc/style.md

```diff
@@ -0,0 +1,274 @@
+# Chisel Style Guide
+
+The Chisel style guide reflects the [Google Java style
+guide](http://https://google.github.io/styleguide/javaguide.html) and the [General Public Scala style
+guide](http://docs.scala-lang.org/style/). Specific rules below are to clarify
+the style used for the chisel3 repo and repos related to Chisel (Firrtl).
+
+**Goal:** Readability and consistency are the main purposes of the style guide.
+Writing your code so someone else (or yourself) can grok it later is important
+to code health and quality.
+
+## Filenames
+The source file name consists of the case-sensitive name of the top-level class
+it contains, plus ".scala".
+
+## Packages
+
+Package definitions must contain the full path to the package from scala. If
+you create a subpackage, it should go in a subdirectory.
+
+    package directory.name.to.get.you.to.your.source
+
+## Imports
+Wildcard ( ._ ) imports are not allowed, with the exception of Chisel._
```

Comment:
I always felt like after a certain number of explicit imports from the same class, it was convenient to use the wildcard. I see the point in avoiding that but its potnetially worth discussing, especially in a case where you only have a small number of imports.